src/a.tex

```diff
@@ -298,19 +320,27 @@ \section{Eventual Success of Store-Conditional Instructions}
 unconditionally fail any unconstrained LR/SC sequence.
 \end{commentary}
 
-If a hart {\em H} enters a constrained LR/SC loop, the execution environment
-must guarantee that one of the following events eventually occurs:
+If each hart in a set of harts is executing a constrained LR/SC loop on the
+same reservation set, at least one hart in that set of harts is guaranteed to
+eventually execute an SC successfully, except under the following conditions:
+\vspace{-0.2in}
+\begin{itemize}
+\parskip 0pt
+\itemsep 1pt
+\item A hart or a device performs a memory write to a byte included in the
+  reservation set.
+\item Any additional conditions specified by any architectural specifications
+  that extend the base specification.
+\end{itemize}
+
+An SC in a constrained LR/SC loop, executed by an individual hart {\em H}, may
+not succeed if any of the following events occurs:
```

Comment:
I’m unsure whether this change (splitting the eventuality concerns into two sections) is an improvement. I find it harder to reason about than the previous monolithic formulation, and it’s not clear to me whether the two are equivalent.

Reply:
Admittedly, I don't believe they are exactly equivalent, but it's also not clear that, if one hart traps or branches out of its loop, the eventual success guarantee should be scrapped.

What we were trying to call out is that 1. there is activity by agents not performing constrained LR/SC loops that can impact those performing said loops, and 2. there are some self-inflicted ways that cause SC failure on one hart, but shouldn't affect the overall guarantee. One could then presume that if 2 happened on all harts executing the loops, then there's no guarantee that one SC would succeed.

In some ways, it's the difference between expressing what guarantees success of one SC (our wording) and what is legal "forward progress" in the EE (original wording, which may include SC failures).

Reply:
@aswaterman Re: "it was perfectly fine for the SC to fail
for no particular reason, as long as it eventually would succeed. If this
new formulation disallows that, it’s a bug in the new formulation."
That wasn't clear to me from the original text, but I understand the intent.
I feel the wording we added specifies what *must* make the SC fail (it's a restatement of the "must fail" sentences in the original text, correct?); however, maybe an explicit statement that an individual SC execution can fail for any reason as long as 8.3 is satisfied is warranted.